bootstrap/kubeadm/api/v1alpha3/kubeadmbootstrapconfig_types.go

```diff
@@ -175,6 +183,71 @@ type File struct {
 	Content string `json:"content"`
 }
 
+// FileSource is a union of all possible external source types for file data.
+// Only one field may be populated in any given instance. Developers adding new
+// sources of data for target systems should add them here.
+type FileSource struct {
+	// Secret represents a secret that should populate this file.
+	// +optional
+	Secret *SecretFileSource `json:"secret,omitempty"`
+}
+
+type FileSourceOptions struct {
+	// DefaultPermissions specifies optional mode bits to use on created files by default.
+	// Must be a value between 0 and 0777.
+	// +optional
+	DefaultPermissions string `json:"defaultPermissions,omitempty"`
+
+	// DefaultOwner specifies the ownership of the file, e.g. "root:root".
+	// +optional
+	DefaultOwner string `json:"defaultOwner,omitempty"`
+
+	// DefaultEncoding specifies the encoding of the file contents.
+	// +optional
+	DefaultEncoding Encoding `json:"defaultEncoding,omitempty"`
+}
```

Comment:
If these are all optional, is there a default for these defaults?

Reply:
cloud-init defaults, will add to doc comments.